fastavro/_schema.pyx

```diff
@@ -0,0 +1,190 @@
+# cython: auto_cpdef=True
+
+from os import path
+import json
+
+PRIMITIVES = set([
+    'boolean',
+    'bytes',
+    'double',
+    'float',
+    'int',
+    'long',
+    'null',
+    'string',
+])
+
+SCHEMA_DEFS = {
+    'boolean': 'boolean',
+    'bytes': 'bytes',
+    'double': 'double',
+    'float': 'float',
+    'int': 'int',
+    'long': 'long',
+    'null': 'null',
+    'string': 'string',
+}
+
+
+class UnknownType(Exception):
+    def __init__(self, name):
+        super(UnknownType, self).__init__(name)
+        self.name = name
+
+
+cpdef inline extract_record_type(schema):
+    if isinstance(schema, dict):
+        return schema['type']
+
+    if isinstance(schema, list):
+        return 'union'
+
+    return schema
+
+
+cpdef inline str extract_logical_type(schema):
+    cdef dict d_schema
+    if not isinstance(schema, dict):
+        return None
+    d_schema = schema
+    rt = d_schema['type']
```

Comment:
Can you update the `schema.py` file to also do `rt = schema['type']` instead of `rt = extract_record_type(schema)` so that they are consistent?

Reply:
Will do.

Reply:
Done

Reply:
Done